src/magicbox/QueryProcessor.ts

```diff
@@ -0,0 +1,99 @@
+export interface IQueryProcessorOptions {
+  timeout: number;
+}
+
+export enum ProcessingStatus {
+  Finished,
+  TimedOut,
+  Overriden
+}
+
+export interface IQueryProcessResult<T> {
+  status: ProcessingStatus;
+  results?: T[];
+}
+
+/**
+ * IE11 equivalent of Promise.race
+ */
+function racePromises<T>(...promises: Thenable<T>[]): Promise<T> {
```

Comment:
@olamothe says he doesn't want me to use a polyfill. However, he recommended I find a more official implementation of it, so that's what I'm going to do.